storage/maria/ma_ft_parser.c

```diff
@@ -86,10 +86,10 @@ my_bool maria_ft_boolean_check_syntax_string(const uchar *str)
   uint i, j;
 
   if (!str ||
-      (strlen((const char *) str) + 1 != sizeof(ft_boolean_syntax)) ||
+      (strlen((const char *) str) + 1 != strlen(ft_boolean_syntax)) ||
```

Comment:
Since string literals are arrays of characters which include terminating `\0`, their `sizeof()` is their byte length plus one. For example, `sizeof("123")` will be 4. That's why there is a `+ 1` on the left side. To make this change correct you need to either remove `+ 1` on the left side, or to add `+ 1` on the right side.

Or you can remove `maria_ft_boolean_check_syntax_string()` completely, since it's not used anywhere.

Reply:
> Since string literals are arrays of characters which include terminating `\0`, their `sizeof()` is their strlen plus one. For example, `sizeof("123")` will be 4. That's why there is a `+ 1` on the left side. To make this change correct you need to either remove `+ 1` on the left side, or to add `+ 1` on the right side.
> 
> Or you can remove `maria_ft_boolean_check_syntax_string()` completely, since it's not used anywhere.

> Hi @i-rinat , thanks for your advice, I have modified my codes.